chapter08/spg/exercise2.pl

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+my $file_handles = {};
+
+open my $input_file, '<', 'datafile.txt' or die "Could not open datafile.txt: $!";
+
+while ( <$input_file> ) {
+  my $line = readline $input_file;

```

Comment:
The diamond operator will read a line from the handle - doesn't this result in `$line` only picking up every second line?  You could have assigned $line in the loop control - i.e. `while ( my $line = <$input_file> ) {`